lib/java/com/google/android/material/button/MaterialButtonHelper.java

```diff
@@ -134,11 +134,10 @@ void loadFromAttributes(@NonNull TypedArray attributes) {
       updateBackground();
     }
     // Set the stored padding values
-    ViewCompat.setPaddingRelative(
-        materialButton,
-        paddingStart + insetLeft,
+    materialButton.setPadding(
```

Comment:
this and below `setPadding` calls will break any existing usages of `paddingStart` and `paddingEnd` in RTL so I don't think we can do that

Reply:
Could you give more details about that?

Reply:
I've written a small test comparing the current and new approach, maybe it will help you to describe the issue:

```java
package com.google.android.material;

import static com.google.common.truth.Truth.assertThat;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.graphics.drawable.ShapeDrawable;
import android.view.View;

import androidx.test.core.app.ApplicationProvider;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

@RunWith(RobolectricTestRunner.class)
public class ViewPaddingTest {

  private final Context context = ApplicationProvider.getApplicationContext();

  @Test
  public void test() {
    View view1 = new View(context);
    View view2 = new View(context);

    view1.setPadding(1, 2, 3, 4);
    view2.setPadding(1, 2, 3, 4);

    currentStoreAndLoadPaddings(view1);
    newStoreAndLoadPaddings(view2);

    assertThat(view2.getPaddingLeft()).isEqualTo(view1.getPaddingLeft());
    assertThat(view2.getPaddingStart()).isEqualTo(view1.getPaddingStart());
    assertThat(view2.getPaddingTop()).isEqualTo(view1.getPaddingTop());
    assertThat(view2.getPaddingRight()).isEqualTo(view1.getPaddingRight());
    assertThat(view2.getPaddingEnd()).isEqualTo(view1.getPaddingEnd());
    assertThat(view2.getPaddingBottom()).isEqualTo(view1.getPaddingBottom());
  }

  /**
   * <a href="https://github.com/material-components/material-components-android/blob/9b6ceac72a084bf187f6e3238dd9283e861196c3/lib/java/com/google/android/material/button/MaterialButtonHelper.java#L123-L141">MaterialButtonHelper.java#L123-L141 from the <code>master</code> branch</a>
   */
  private void currentStoreAndLoadPaddings(View view) {
    int paddingStart = view.getPaddingStart();
    int paddingTop = view.getPaddingTop();
    int paddingEnd = view.getPaddingEnd();
    int paddingBottom = view.getPaddingBottom();

    view.setBackground(createTestDrawableWithPaddings());

    view.setPaddingRelative(paddingStart, paddingTop, paddingEnd, paddingBottom);
  }

  /**
   * <a href="https://github.com/material-components/material-components-android/blob/c06f058d7c478b82aaf5c7eddce0a3a54ce3f50c/lib/java/com/google/android/material/button/MaterialButtonHelper.java#L123-L141">MaterialButtonHelper.java#L123-L141 from the PR</a>
   */
  private void newStoreAndLoadPaddings(View view) {
    int paddingLeft = view.getPaddingLeft();
    int paddingTop = view.getPaddingTop();
    int paddingRight = view.getPaddingRight();
    int paddingBottom = view.getPaddingBottom();

    view.setBackground(createTestDrawableWithPaddings());

    view.setPadding(paddingLeft, paddingTop, paddingRight, paddingBottom);
  }

  private Drawable createTestDrawableWithPaddings() {
    ShapeDrawable testDrawable = new ShapeDrawable();
    testDrawable.setPadding(10, 11, 12, 13);
    return testDrawable;
  }
}
```